File: 152-maximum-product-subarray/maximum-product-subarray.cpp
class Solution {
public:
    int maxProduct(vector<int>& nums) {
        int result = nums[0];
        int minv = 1, maxv = 1;
        for (int n: nums){
            int temp = maxv;
            maxv = max({maxv * n, minv * n, n});
            minv = min({temp * n, minv * n, n});
            result = max(result, maxv);
        }
        return result;
    }
};

File: 152-maximum-product-subarray/maximum-product-subarray.py
class Solution:
    def maxProduct(self, nums: List[int]) -> int:
        result = max(nums)
        maxv, minv = 1, 1

        for i in nums:
            temp = maxv
            maxv = max(maxv * i, minv * i, i)
            minv = min(temp * i, minv * i, i)
            result = max(result, maxv)
        return result